typing/jkind.ml

```diff
@@ -856,16 +884,24 @@ module Layout_and_axes = struct
           loop ctl bounds_so_far bs
         | false -> (
           let join_bounds b1 b2 ~relevant_axes =
-            Mod_bounds.Map2.f
-              { f =
-                  (fun (type a) ~(axis : a Axis.t) b1 b2 ->
-                    if Axis_set.mem relevant_axes axis
-                    then
-                      let (module Bound_ops) = Axis.get axis in
-                      Bound_ops.join b1 b2
-                    else b1)
-              }
-              b1 b2
+            let value_for_axis (type a) ~(axis : a Axis.t) : a =
+              if Axis_set.mem relevant_axes axis
+              then
+                let (module Bound_ops) = Axis.get axis in
```

Comment:
Can we do better here? I thought we identified this as an allocate-y operation which might be slow. And it's in a loop here. It's a bit more verbose, but https://github.com/goldfirere/flambda-backend/blob/rae/dismantle-axis-collection/typing/jkind.ml#L1654 has another way.

Reply:
We could also make `value_for_axis` take the join operation as an argument.

Reply:
this doesn't end up showing up on a profile, fwiw, for whatever reason. We could make it worse for consistency's sake, but I'm hesitant to?

Reply:
`Accent_lattice` was being referenced from `get_max_axes`, not here - that's gone now, and this is now below threshold on a profile.